libraries/rush-lib/src/logic/base/BaseInstallManager.ts

```diff
@@ -154,11 +177,26 @@ export abstract class BaseInstallManager {
     return this._options;
   }
 
+  protected get commonTempInstallFlag(): LastInstallFlag {
+    return this._commonTempInstallFlag;
+  }
+
+  protected get twoStageInstall(): boolean {
+    return this._twoStageInstall;
+  }
+
   public async doInstallAsync(): Promise<void> {
     const isFilteredInstall: boolean = this.options.pnpmFilterArguments.length > 0;
     const useWorkspaces: boolean =
       this.rushConfiguration.pnpmOptions && this.rushConfiguration.pnpmOptions.useWorkspaces;
 
+    this._twoStageInstall = useWorkspaces && this.rushConfiguration.packageManager === 'pnpm';
```

Comment:
For the non-draft PR, we should provide a setting to restore the old behavior, in case people encounter any problems with this new mode.  

An optimistic approach might be `legacyPostinstallBehavior=true` (opt-out) in **experiments.json**.  

A less optimistic approach might be `deferredPostinstall=true` (opt-in) in **experiments.json**.

@elliot-nelson @D4N14L do you have opinions on naming/optimism?

Reply:
@chengcyber pointed out that `postinstall` doesn't have any special status among the sequence of `preinstall`, `install`, `postinstall`, `preprepare` ...  So maybe we can call them "installation scripts" (following the naming of `pnpm install --ignore-scripts`).

We discussed it a bit and liked the names  `deferredInstallationScripts` or `runInstallationScriptsSeparately`. I don't have a strong opinion.

Reply:
renaming "twoStageInstall" to "deferredInstallationScripts".